### PR TITLE
Balancer Impl

### PR DESCRIPTION
**Pull request by Drypto13**

### CODE REVIEW

File: contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol [hunk 1]
```diff
@@ -165,11 +165,13 @@ contract SwapsImplBalancer_POLYGON is State, ISwapsImpl {
             (IBalancerVault.BatchSwapStep[] memory swapParams, address[] memory tokens, int256[] memory limits) = abi.decode(payload, (IBalancerVault.BatchSwapStep[], address[], int256[]));
             require(tokens[0] == sourceTokenAddress && tokens[tokens.length-1] == destTokenAddress, "invalid tokens");
             limits[0] = int256(minSourceTokenAmount);
+            require(limits[limits.length-1] <= 0, "cannot spend dest token");
```
Comment (0xosama) [inline contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol]: can you extract this into a function `checkLimits` or something

Comment (Drypto13) [reply contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol]: completed

File: contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol [hunk 2]
```diff
@@ -165,11 +172,10 @@ contract SwapsImplBalancer_POLYGON is State, ISwapsImpl {
             (IBalancerVault.BatchSwapStep[] memory swapParams, address[] memory tokens, int256[] memory limits) = abi.decode(payload, (IBalancerVault.BatchSwapStep[], address[], int256[]));
             require(tokens[0] == sourceTokenAddress && tokens[tokens.length-1] == destTokenAddress, "invalid tokens");
```
Comment (yanga128) [inline contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol:173]: this require and   _checkLimits(limits) can be moved above the condition, they are the same in both cases

Comment (yanga128) [reply contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol:173]: _checkLimits has loop inside and there is more and less the same loop right after _checkLimits call, maybe it's possible to combine and loop only once

Comment (Drypto13) [reply contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol:173]: this has been adjusted

**Review by 0xosama — APPROVED** [review]
looks good

**Pull request merged**